test_communication/test/requester_py.py

```diff
@@ -32,6 +32,11 @@ def requester(service_name, number_of_cycles):
     srv_fixtures = get_test_srv(service_name)
 
     client = node.create_client(srv_mod, 'test_service_' + service_name)
+    tries = 15
+    while rclpy.ok() and not client.wait_for_service(timeout_sec=1.0) and tries > 0:
+        print('service not available, waiting again...')
+        tries -= 1
+    assert tries > 0
```

Comment:
We actually need to cleanup/shutdown here before exiting.
```
     node.destroy_node()
     rclpy.shutdown()
```

Reply:
@mikaelarguedas refactored to make sure cleanup happens in cb8ce48cff29dec596249db096b9dc29b16af6b3

I removed `--number_of_cycles`/`spin_count` since they were broken and unused. The while loop only executed once because of a `break` statement. `grep` in `system_tests` for `number_of_cycles` and didn't find anything using it.

Reply:
@sloretz indeed number of cycles was not used here :+1: I guess it was a remnant of the original copy-n-paste from the publisher subscriber.